syntax="proto3";

package distributed_agenda;
option go_package = "./server/proto";

import "proto/users.proto";

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  string token = 1;
}

message SignUpRequest{
  User user = 1;
}

message SignUpResponse{
  OperationOutcome result = 1;
}

service Auth {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc SignUp (SignUpRequest) returns (SignUpResponse);
}